ui/desktop/app/controllers/scopes/scope/projects/targets/index.js

```diff
@@ -21,15 +21,36 @@ export default class ScopesScopeProjectsTargetsIndexController extends Controlle
 
   // =attributes
 
-  queryParams = ['search', 'page', 'pageSize'];
+  queryParams = ['search', { scopes: { type: 'array' } }, 'page', 'pageSize'];
 
+  allTargets;
   @tracked search;
+  @tracked scopeSearchTerm;
+  @tracked scopes = [];
+  @tracked selectedScopes = [];
   @tracked page = 1;
   @tracked pageSize = 10;
-  @tracked searchItems = ['Project 1', 'Project 2', 'Project 3'];
+
+  constructor() {
+    super(...arguments);
+    this.getAllTargets();
+  }
 
   // =methods
 
+  async getAllTargets() {
+    // Query all targets for defining filtering values
+    const options = { pushToStore: false };
+    let allTargets = await this.store.query('target', { query: null }, options);
+
+    // Filter out targets to which users do not have the connect ability
+    allTargets = allTargets.filter((target) =>
+      target.authorized_actions.includes('authorize-session'),
```

Comment:
can this be combined with the query method? 

`const options = { pushToStore: false, authorized_actions: ['authorize-session'] };`

Reply:
Not currently but you can ask backend to see if it's something that is possible to be searched on

Reply:
I am not sure if we need to. I am only doing this because I don't believe I can use the `can` abilities without it being a TargetModel.